src/box/sql/os_unix.c

```diff
@@ -1437,10 +1437,16 @@ unixTempFileDir(void)
 	struct stat buf;
 	const char *zDir = sql_temp_directory;
 
-	if (!azDirs[0])
+	if (!azDirs[0]) {
 		azDirs[0] = getenv("SQL_TMPDIR");
-	if (!azDirs[1])
+		if (azDirs[0] != 0 && strnlen(azDirs[0], PATH_MAX) >= PATH_MAX)
```

Comment:
Let's add a helper `getenv_safe(name, max_len)` and use it everywhere instead of `getenv`?

I think we could add a helper like this for Lua as well.

Reply:
Done.